xml/telnet_banners.xml

```diff
@@ -0,0 +1,820 @@
+      DgvMDcvMTAgKFNWTiByZXZpc2lvbjogMTQ4OTYpDQoNClByb2xpYW50IERMOTgwUjA3IFg2NT
+      UwIDgtY29yZSA0UCBTQVMgbG9naW46Cg==
+    </example>
+    <param pos="0" name="os.vendor" value="DD-WRT"/>
+    <param pos="0" name="os.product" value="DD-WRT"/>
+    <param pos="0" name="os.device" value="Router"/>
+    <param pos="1" name="os.version"/>
+    <param pos="2" name="os.version.version"/>
+    <param pos="3" name="os.edition"/>
+    <param pos="4" name="os.build"/>
+  </fingerprint>
+  <fingerprint pattern="^(?:\r|\n)*DD-WRT v(3.\d)-(r([\w]+)) ([\w-]+) \(c\) \d{4} NewMedia-NET GmbH(?:\r|\n)+Release: \d+\/\d+\/\d+(?:\r|\n)+.* login:\s*$">
+    <description>DD-WRT - 3.0 family</description>
+    <!-- DD-WRT v3.0-r34886M std (c) 2018 NewMedia-NET GmbH\r\nRelease: 02/10/18\r\n\r\nwibrate login: -->
+    <example _encoding="base64" os.version="3.0" os.version.version="r34886M" os.edition="std" os.build="34886M">
+      REQtV1JUIHYzLjAtcjM0ODg2TSBzdGQgKGMpIDIwMTggTmV3TWVkaWEtTkVUIEdtYkgNClJlb
+      GVhc2U6IDAyLzEwLzE4DQoNCndpYnJhdGUgbG9naW46Cg==
+    </example>
+    <param pos="0" name="os.vendor" value="DD-WRT"/>
+    <param pos="0" name="os.product" value="DD-WRT"/>
+    <param pos="0" name="os.device" value="Router"/>
+    <param pos="1" name="os.version"/>
+    <param pos="2" name="os.version.version"/>
+    <param pos="3" name="os.build"/>
+    <param pos="4" name="os.edition"/>
+  </fingerprint>
+  <fingerprint pattern="^(TD-\w+) [\d.]+ DSL Modem Router(?:\r|\n)+Authorization failed after trying \d+ times!!!\.(?:\r|\n)+Please login after \d+ seconds!\s*$">
+    <description>TP-LINK TD Family DSL Modem/Router</description>
+    <!-- TD-W8960N 5.0 DSL Modem Router\r\nAuthorization failed after trying 5 times!!!.\r\nPlease login after 416 seconds! -->
+    <example _encoding="base64" hw.product="TD-W8960N">
+      VEQtVzg5NjBOIDUuMCBEU0wgTW9kZW0gUm91dGVyDQpBdXRob3JpemF0aW9uIGZhaWxlZCBhZ
+      nRlciB0cnlpbmcgNSB0aW1lcyEhIS4NClBsZWFzZSBsb2dpbiBhZnRlciA0MTYgc2Vjb25kcy
+      E=
+    </example>
+    <param pos="0" name="hw.vendor" value="TP-Link"/>
+    <param pos="1" name="hw.product"/>
+    <param pos="0" name="hw.device" value="Router"/>
+  </fingerprint>
+  <fingerprint pattern="^(?:\r|\n)*ZyXEL login:$">
+    <description>ZyXEL simple</description>
+    <example>ZyXEL login:</example>
+    <param pos="0" name="service.vendor" value="ZyXEL"/>
+    <param pos="0" name="hw.vendor" value="ZyXEL"/>
+  </fingerprint>
+  <fingerprint pattern="^ZyXEL \w?DSL Router\r\nLogin:$">
+    <description>ZyXEL Router - simple</description>
+    <!-- ZyXEL VDSL Router\r\nLogin: -->
+    <example _encoding="base64">WnlYRUwgVkRTTCBSb3V0ZXINCkxvZ2luOgo=</example>
+    <param pos="0" name="service.vendor" value="ZyXEL"/>
+    <param pos="0" name="hw.vendor" value="ZyXEL"/>
+    <param pos="0" name="hw.device" value="Router"/>
+  </fingerprint>
+  <fingerprint pattern="^Debian GNU\/Linux 9(?:\r|\n)+([\w.-]+) login:\s*$">
```

Comment:
The next two examples are missing the leading carriage return and newline part of the fingerprint.

Reply:
I found no examples of particular banners with leading CR/LF.

Reply:
Ah, interesting.  For some reason I had assumed that all regexes in here were modified to account for the possible whitespace at the beginning of the banner.

Reply:
No, the current leading regex for CR/LF is left over from my pre-trim-the-data work.  It should only be present when I actually saw banners with leading/trailing CR, LF, nulls, and other whitespace.